pkg/daemon/vm_controller.go

```diff
@@ -55,7 +68,9 @@ func (r *VMReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Re
 }
 
 func (r *VMReconciler) reconcile(ctx context.Context, vm *virtv1alpha1.VirtualMachine) error {
-	if vm.Status.NodeName == "" || vm.Status.NodeName != r.NodeName {
+	shouldReconcile := vm.Status.NodeName != "" && vm.Status.NodeName == r.NodeName ||
+		vm.Status.Migration != nil && vm.Status.Migration.TargetNodeName != "" && vm.Status.Migration.TargetNodeName == r.NodeName
+	if !shouldReconcile {
```

Comment:
Use `()` when mixing `&&` and `||`

Reply:
And maybe we can move this logic inside each case?

Reply:
this is the common logic for all the below codes.  put it at top make codes clear.

Reply:
Actually, depending on whether the VM is in a migration, this if condition varies. If we put it inside each case, each condition can be simple and very readable. Now this if condition is too complicated and hard to read.

Reply:
maybe return early is a good practice here, the daemon only handle VM currently on this node or migrating to this node, if put this logic to below, there are many sub-branch, will repeat and maybe miss.

Reply:
It's too hard to read and understand. Or at least we should use `()` to make the condition more clear